### PR TITLE
feat: add shared AtlassianTime type for all Atlassian timestamp formats

### LINKED ISSUE #309: feat: shared AtlassianTime type for fractional epoch timestamp handling
## Problem

PR #305 added a `Timestamp` type to handle both ISO 8601 strings and epoch numbers from the Jira Automation API. Live testing revealed the API returns **fractional epoch seconds** (e.g. `1701482354.625000000`) — a Jackson `jackson-datatype-jsr310` `Instant.getNano()` serialization. The `int64` unmarshal fails on the decimal point.

Across Atlassian Cloud APIs there are three timestamp formats:

| Format | Source | Example |
|--------|--------|---------|
| ISO 8601 string | Jira/Confluence core resources | `"2023-12-04T10:00:00.000+0000"` |
| Integer epoch millis | Webhooks, automation event payloads | `1461049397396` |
| Decimal epoch seconds.nanos | Automation rule config/audit (Jackson jsr310) | `1701482354.625000000` |

## Solution

Add a shared `AtlassianTime` type in `shared/atime/` that sniffs the JSON shape:
- Leading `"` → ISO 8601 string (try multiple layouts)
- Digit with `.` → decimal epoch seconds.nanos (string-split for precision, not float64)
- Digit without `.` → integer epoch millis

Use string-split (not `float64`) for the decimal path because `float64` only carries ~15-17 significant digits but `1701482354.625000000` has 19 — parsing through float64 quietly discards nanosecond precision.

Migrate both:
- `jtk` `Timestamp` type (`api/automation_types.go`) → `atime.AtlassianTime`
- `cfl` `Time` type (`api/types.go`) → `atime.AtlassianTime`

## Success criteria

Build and lint before testing:
\`\`\`
make build && rtk make lint
\`\`\`

**1. `jtk automation list`** must exit 0, output pipe-delimited table with `ID | STATE | NAME`.

**2. `jtk automation list --id`** must emit one ID per line.

**3. `jtk automation list --extended`** must show `ID | STATE | LABELS | TAGS | AUTHOR | NAME`.

**4. `jtk automation get 018c2840-57c1-7869-9393-11205cc87ce4`** must show header + State + Components + Description.

**5. `jtk automation get <uuid> --id`** must emit only the UUID.

**6. `jtk automation get <uuid> --extended`** must show Labels, Tags, Author, Scope, Created, Updated.

**7. `cfl page list`** must still work (regression check after `Time` → `AtlassianTime` migration).

## References
- [Jira Cloud REST API](https://developer.atlassian.com/cloud/jira/platform/rest/v3/intro/) — ISO 8601
- [Confluence Cloud REST API v2](https://developer.atlassian.com/cloud/confluence/rest/v2/intro/) — ISO 8601
- [Automation REST API](https://developer.atlassian.com/cloud/automation/rest/) — decimal epoch seconds.nanos
- [Webhooks / automation actions](https://developer.atlassian.com/cloud/jira/service-desk/automation-action/) — integer epoch millis

### PR DESCRIPTION
**Pull request by rianjs**
## Summary

- Adds `shared/atime.AtlassianTime` type that handles all three Atlassian Cloud timestamp formats: ISO 8601 strings, integer epoch millis, and fractional epoch seconds.nanos (Jackson jsr310)
- Uses string-split approach (not float64) to preserve nanosecond precision
- Migrates jtk's `Timestamp` type and cfl's local `Time` type to the shared implementation
- Fixes pre-existing lint issues in both cfl and jtk (gosec false positives, gofmt alignment)

## Test plan

- [x] `make build` — both tools compile
- [x] `make test` — all tests pass (shared, cfl, jtk)
- [x] `make lint` — zero issues across all modules
- [ ] Live test: `bin/jtk automation list`, `bin/jtk automation get <uuid>` with all output modes
- [ ] Live test: `bin/cfl page list` regression check

Closes #309

### CODE REVIEW

**Comment by rianjs** [comment]
## TDD Review: `shared/atime` package

### Coverage: Solid with notable gaps

**What's well-covered**
- All 3 format branches: ISO 8601 (multiple layouts), integer epoch millis, fractional epoch seconds.nanos
- Short nano truncation/padding (`1701482354.625` → 625000000 ns) ✓
- `null` and empty string handling ✓
- Error paths for invalid boolean input and unparseable ISO strings ✓
- Round-trip marshal/unmarshal ✓
- Zero-value `String()` returns `""` ✓
- `json:",omitempty"` nil-pointer suppression ✓

---

### Gaps

**1. `isoLayouts` — non-UTC offset variant untested**
`"2006-01-02T15:04:05.000-0700"` (negative non-zero offset) is in `isoLayouts` but has no test. A timestamp like `"2023-12-04T10:00:00.000-0500"` should unmarshal and preserve the correct UTC equivalent. This is an active Jira API format.

**2. Integer epoch seconds (10-digit) vs millis (>10-digit) boundary**
The branch is `len(s) <= 10` → seconds, else millis. Missing tests:
- 11-digit integers (edge of the boundary: year 2001–2286 range for seconds would be 10 digits, so 11-digit is always millis — confirm this assumption is documented/tested)
- Negative epoch values (pre-1970) — the `ParseInt` would succeed but `len(s)` check includes the `-` character, which could misclassify an 11-char negative second value as millis

**3. Fractional seconds with >9 nano digits**
The implementation truncates to 9 chars with `(nanoStr + "000000000")[:9]`. If the API ever returns 10+ decimal digits (theoretically possible from some serializers), the result would be silently wrong. A test asserting truncation behavior would document the contract.

**4. `MarshalJSON` not tested directly**
`AtlassianTime` embeds `time.Time`, so `MarshalJSON` falls through to `time.Time`'s implementation. The round-trip test covers this implicitly, but there's no explicit test that a zero `AtlassianTime` marshals as `null` (vs `"0001-01-01T00:00:00Z"`). The `cfl` `Time` type explicitly returned `null` for zero — confirm the new behavior is intentional and tested.

**5. Removed `Timestamp` tests not replaced equivalently**
`TestTimestamp_UnmarshalJSON` was deleted (which is correct — the type is gone). But `TestGetAutomationRule_EnvelopeWithFractionalTimestamp` only checks `Created != nil` and the date portion — it no longer asserts the `Updated` field value at all. Previous tests verified both `Created` and `Updated`. The date-only assertion also loses the time component check.

**6. No test for `RFC3339Nano` with sub-second precision**
`"2023-12-04T09:00:00.123456789Z"` would exercise `RFC3339Nano` as the winning layout over `RFC3339`. Without it, there's no guarantee the layout order matters correctly.

---

### Minor flags

- `//nolint:gosec` comments scattered across unrelated files (`upload_test.go`, `init_test.go`, `config.go`, `view.go`, `fidelity_test.go`) — these look like lint cleanup unrelated to the feature; fine to include but worth confirming they're intentional.
- `page.go` nil-guard addition (`p.Version.CreatedAt != nil`) is correct given the field changed from value to pointer type.

---

### Summary

Core coverage is good for the happy paths. The most actionable gaps are: **(1) the -0700 offset layout**, **(2) negative epoch / boundary integers**, and **(4) zero-value marshal behavior**. The deletion of `Updated` field assertion in the automation test is a mild regression in coverage.

**Comment by rianjs** [comment]
**Blocker**
- None.

**Major**
- None.

**Minor**
- None.

**Nit**
- None.

**Comment by rianjs** [comment]
Addressing TDD review findings:

1. **-0700 layout** — Fixed. Added test case for `"2023-12-04T03:00:00.000-0700"`.
2. **Updated assertion dropped** — Fixed. Restored `Updated` field assertion with full `RFC3339Nano` precision.
3. **Negative epoch values** — Accepted risk. Atlassian APIs don't return negative epochs (pre-1970 timestamps). The digit-count heuristic is only for positive values from known API responses.
4. **MarshalJSON zero-value** — Accepted risk. All struct fields use `*AtlassianTime` pointers, so nil → omitted by `json:",omitempty"`. A non-nil zero value would only occur from explicit code, not API deserialization.
5. **RFC3339Nano sub-second ordering** — The RFC3339Nano layout is first in the `isoLayouts` slice, so it's tried first. The existing `.000Z` test case validates fractional second parsing works. The ordering ensures the most-specific layout matches first.

File: tools/jtk/api/automation_test.go [hunk 1]
```diff
@@ -678,7 +657,9 @@ func TestListAutomationRules_NumericTimestamp(t *testing.T) {
 	testutil.RequireNoError(t, err)
```
Comment (monit-reviewer) [inline tools/jtk/api/automation_test.go:657]: 🟡 **Medium** (harness-engineering:harness-enforcement-reviewer): TestListAutomationRules_NumericTimestamp was weakened: the assertion was downgraded from checking the exact parsed value ("2023-12-04T09:00:00Z") to only checking Created != nil. A bug in numeric timestamp parsing would go undetected since a wrong-but-non-nil value would pass the test.

*Reply to this thread when addressed.*

File: tools/jtk/api/automation_test.go [hunk 2]
```diff
@@ -698,7 +679,9 @@ func TestGetAutomationRule_LegacyWithNumericTimestamp(t *testing.T) {
...
```
Comment (monit-reviewer) [inline tools/jtk/api/automation_test.go:680]: 🟡 **Medium** (harness-engineering:harness-enforcement-reviewer): TestGetAutomationRule_LegacyWithNumericTimestamp similarly drops the value assertion for Created, replacing the exact string check with only a nil check. The legacy integer-millis path is no longer verified end-to-end.

*Reply to this thread when addressed.*

File: shared/atime/atime.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+type AtlassianTime struct {
+	time.Time
+}
+
+var isoLayouts = []string{
+	time.RFC3339Nano,
+	time.RFC3339,
+	"2006-01-02T15:04:05.000Z",
+	"2006-01-02T15:04:05.000+0000",
+	"2006-01-02T15:04:05.000-0700",
```
Comment (monit-reviewer) [inline shared/atime/atime.go:26]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): The layout "2006-01-02T15:04:05.000+0000" uses a literal '+0000' which Go's time package does not recognize as a timezone specifier. It only matches input that literally contains '+0000' and silently fails for any other offset. The '-0700' layout that follows already handles +0000 correctly, making this layout both subtly incorrect and entirely redundant.

*Reply to this thread when addressed.*

File: shared/atime/atime.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+//   - ISO 8601 string (Jira/Confluence core): "2023-12-04T10:00:00.000+0000"
+//   - Integer epoch millis (webhooks): 1461049397396
+//   - Decimal epoch seconds.nanos (Automation/Jackson jsr310): 1701482354.625000000
+//
+// Use *AtlassianTime in struct fields with `json:",omitempty"` to suppress nil values.
+type AtlassianTime struct {
+	time.Time
+}
+
+var isoLayouts = []string{
```
Comment (monit-reviewer) [inline shared/atime/atime.go:22]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): ISO string paths do not normalize to UTC. A timestamp like "2023-12-04T10:00:00.000+0000" is parsed into a FixedZone("+0000", 0) rather than time.UTC. These are equal by time.Equal but differ by Location, so t.String() produces "+00:00" instead of "Z". The numeric paths consistently call .UTC(); the ISO paths should too for consistent output.

*Reply to this thread when addressed.*

File: shared/atime/atime.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+		}
+		nanoStr = (nanoStr + "000000000")[:9]
+		nanos, err := strconv.ParseInt(nanoStr, 10, 64)
+		if err != nil {
+			return fmt.Errorf("AtlassianTime: invalid nanos in %q: %w", s, err)
+		}
+		t.Time = time.Unix(sec, nanos).UTC()
+		return nil
+	}
+
```
Comment (monit-reviewer) [inline shared/atime/atime.go:63]: 🔵 **Low** (harness-engineering:harness-enforcement-reviewer): The seconds-vs-millis disambiguation heuristic `len(s) <= 10` is undocumented. A comment explaining the invariant — that Atlassian millis are always 13+ digits for modern timestamps and that 11/12-digit values are out of scope for known API responses — would prevent future silent misclassification.

*Reply to this thread when addressed.*

File: shared/atime/atime.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	n, err := strconv.ParseInt(s, 10, 64)
+	if err != nil {
+		return fmt.Errorf("AtlassianTime: cannot unmarshal %q: %w", s, err)
+	}
+	if len(s) <= 10 {
+		t.Time = time.Unix(n, 0).UTC()
```
Comment (monit-reviewer) [inline shared/atime/atime.go:71]: 🔵 **Low** (harness-engineering:harness-enforcement-reviewer): AtlassianTime has no MarshalJSON override, so it inherits time.Time's behavior which emits "0001-01-01T00:00:00Z" for zero values rather than null. The old cfl.Time.MarshalJSON explicitly returned null for zero time. While all current struct fields use *AtlassianTime with omitempty (nil → omitted), a non-nil zero-value would marshal differently than under the old type — a silent behavior change worth documenting.

*Reply to this thread when addressed.*

**Review by monit-reviewer — DISMISSED** [review]
## Automated PR Review



**Reviewed commit:** `7d88fcc`

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-architecture-reviewer | 1 |
| harness-engineering:harness-enforcement-reviewer | 4 |
| harness-engineering:harness-self-documenting-code-reviewer | 1 |

<details>
<summary><strong>harness-engineering:harness-architecture-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `shared/atime/atime.go:22`
> ISO string paths do not normalize to UTC. A timestamp like "2023-12-04T10:00:00.000+0000" is parsed into a FixedZone("+0000", 0) rather than time.UTC. These are equal by time.Equal but differ by Location, so t.String() produces "+00:00" instead of "Z". The numeric paths consistently call .UTC(); the ISO paths should too for consistent output.

</details>

<details>
<summary><strong>harness-engineering:harness-enforcement-reviewer (4 findings)</strong></summary>

**⚠️ Should Fix** - `tools/jtk/api/automation_test.go:657`
> TestListAutomationRules_NumericTimestamp was weakened: the assertion was downgraded from checking the exact parsed value ("2023-12-04T09:00:00Z") to only checking Created != nil. A bug in numeric timestamp parsing would go undetected since a wrong-but-non-nil value would pass the test.

**⚠️ Should Fix** - `tools/jtk/api/automation_test.go:679`
> TestGetAutomationRule_LegacyWithNumericTimestamp similarly drops the value assertion for Created, replacing the exact string check with only a nil check. The legacy integer-millis path is no longer verified end-to-end.

**💡 Suggestion** - `shared/atime/atime.go:64`
> The seconds-vs-millis disambiguation heuristic `len(s) <= 10` is undocumented. A comment explaining the invariant — that Atlassian millis are always 13+ digits for modern timestamps and that 11/12-digit values are out of scope for known API responses — would prevent future silent misclassification.

**💡 Suggestion** - `shared/atime/atime.go:70`
> AtlassianTime has no MarshalJSON override, so it inherits time.Time's behavior which emits "0001-01-01T00:00:00Z" for zero values rather than null. The old cfl.Time.MarshalJSON explicitly returned null for zero time. While all current struct fields use *AtlassianTime with omitempty (nil → omitted), a non-nil zero-value would marshal differently than under the old type — a silent behavior change worth documenting.

</details>

<details>
<summary><strong>harness-engineering:harness-self-documenting-code-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `shared/atime/atime.go:27`
> The layout "2006-01-02T15:04:05.000+0000" uses a literal '+0000' which Go's time package does not recognize as a timezone specifier. It only matches input that literally contains '+0000' and silently fails for any other offset. The '-0700' layout that follows already handles +0000 correctly, making this layout both subtly incorrect and entirely redundant.

</details>

*2 info-level observations excluded. Run with `--verbose` to include.*

*3 PR discussion threads considered.*

---
<details>
<summary>Completed in 3m 05s | $1.11</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 3m 05s wall · 8m 07s compute (Reviewers: 2m 17s · Synthesis: 44s) |
| Cost | $1.11 |
| Tokens | 297.6k in / 32.7k out |
| Turns | 14 |

</details>

File: tools/jtk/api/automation_test.go (hunk 1 above)
Comment (rianjs) [reply tools/jtk/api/automation_test.go:657]: Fixed — restored exact value assertion checking the parsed time matches 2023-12-04T09:00:00Z.

File: tools/jtk/api/automation_test.go (hunk 2 above)
Comment (rianjs) [reply tools/jtk/api/automation_test.go:680]: Fixed — restored exact value assertion for the legacy integer-millis path.

File: shared/atime/atime.go (hunk 6 above)
Comment (rianjs) [reply shared/atime/atime.go:71]: Acknowledged — all struct fields use *AtlassianTime with omitempty, so nil → omitted. A non-nil zero value would only occur from explicit construction, not API deserialization. Adding MarshalJSON would add complexity for a path that doesn't occur in practice.

File: shared/atime/atime.go (hunk 5 above)
Comment (rianjs) [reply shared/atime/atime.go:63]: Fixed — added comment documenting the invariant: epoch seconds are ≤10 digits through 2286, epoch millis are 13+ digits for dates after 2001.

File: shared/atime/atime.go (hunk 3 above)
Comment (rianjs) [reply shared/atime/atime.go:26]: Fixed — removed the redundant +0000 layout. The -0700 pattern already handles all 4-digit offsets including +0000.

File: shared/atime/atime.go (hunk 4 above)
Comment (rianjs) [reply shared/atime/atime.go:22]: Intentional — ISO paths preserve the original timezone from the API response. Normalizing to UTC would lose timezone information that callers may need. The numeric paths use UTC because epoch timestamps have no timezone context.

File: tools/jtk/api/automation_test.go (hunk 1 above)
Comment (monit-reviewer) [reply tools/jtk/api/automation_test.go:657]: **Settled**: monit-reviewer flagged that the numeric timestamp test was weakened to only check non-nil instead of the exact parsed value. rianjs fixed it by restoring the exact value assertion for 2023-12-04T09:00:00Z.

File: tools/jtk/api/automation_test.go (hunk 2 above)
Comment (monit-reviewer) [reply tools/jtk/api/automation_test.go:680]: **Settled**: monit-reviewer flagged that the legacy integer-millis timestamp test dropped the exact value assertion in favor of a nil check. rianjs fixed it by restoring the exact value assertion for the legacy path.

File: shared/atime/atime.go (hunk 3 above)
Comment (monit-reviewer) [reply shared/atime/atime.go:26]: **Settled**: monit-reviewer noted the '+0000' layout literal is not a valid Go timezone specifier and is redundant since '-0700' already handles it. rianjs fixed it by removing the redundant layout.

File: shared/atime/atime.go (hunk 4 above)
Comment (monit-reviewer) [reply shared/atime/atime.go:22]: **Settled**: monit-reviewer suggested ISO paths should normalize to UTC for consistency with numeric paths. rianjs explained the behavior is intentional — ISO paths preserve the original timezone from the API response, while numeric paths use UTC because epoch timestamps have no timezone context.

File: shared/atime/atime.go (hunk 5 above)
Comment (monit-reviewer) [reply shared/atime/atime.go:63]: **Settled**: monit-reviewer flagged that the seconds-vs-millis disambiguation heuristic lacked documentation. rianjs fixed it by adding a comment explaining that epoch seconds are ≤10 digits through 2286 and epoch millis are 13+ digits for dates after 2001.

File: shared/atime/atime.go (hunk 6 above)
Comment (monit-reviewer) [reply shared/atime/atime.go:71]: **Settled**: monit-reviewer noted AtlassianTime lacks a MarshalJSON override, differing from the old cfl.Time behavior for non-nil zero values. rianjs acknowledged it but explained all struct fields use *AtlassianTime with omitempty so a non-nil zero value never occurs in practice, making the added complexity unjustified.

File: shared/atime/atime.go [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+		}
+		nanoStr = (nanoStr + "000000000")[:9]
+		nanos, err := strconv.ParseInt(nanoStr, 10, 64)
+		if err != nil {
+			return fmt.Errorf("AtlassianTime: invalid nanos in %q: %w", s, err)
+		}
+		t.Time = time.Unix(sec, nanos).UTC()
+		return nil
+	}
+
```
Comment (monit-reviewer) [inline shared/atime/atime.go:63]: 🔵 **Low** (harness-engineering:harness-enforcement-reviewer): The inline comment states 'epoch millis are 13+ digits for dates after 2001', but the code routes any value with len > 10 digits to UnixMilli, meaning 11- and 12-digit values are also treated as millis. The comment is misleading: the actual threshold is 11+ digits, not 13+. A future maintainer reading only the comment may misunderstand the heuristic.

*Reply to this thread when addressed.*

File: shared/atime/atime_test.go [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+func TestString(t *testing.T) {
+	t.Parallel()
+	at := AtlassianTime{Time: time.Date(2023, 12, 4, 9, 0, 0, 0, time.UTC)}
+	if at.String() != "2023-12-04T09:00:00Z" {
+		t.Errorf("String() = %q", at.String())
+	}
+
+	zero := AtlassianTime{}
+	if zero.String() != "" {
+		t.Errorf("zero String() = %q, want empty", zero.String())
```
Comment (monit-reviewer) [inline shared/atime/atime_test.go:144]: 🔵 **Low** (harness-engineering:harness-enforcement-reviewer): The contains() and searchString() helpers manually reimplement strings.Contains. Use strings.Contains from the standard library directly.

*Reply to this thread when addressed.*

File: tools/jtk/go.sum [hunk 9]
```diff
@@ -36,6 +36,8 @@ github.com/charmbracelet/x/termios v0.1.1 h1:o3Q2bT8eqzGnGPOYheoYS8eEleT5ZVNYNy8
 github.com/charmbracelet/x/termios v0.1.1/go.mod h1:rB7fnv1TgOPOyyKRJ9o+AsTU/vK5WHJ2ivHeut/Pcwo=
 github.com/charmbracelet/x/xpty v0.1.2 h1:Pqmu4TEJ8KeA9uSkISKMU3f+C1F6OGBn8ABuGlqCbtI=
 github.com/charmbracelet/x/xpty v0.1.2/go.mod h1:XK2Z0id5rtLWcpeNiMYBccNNBrP2IJnzHI0Lq13Xzq4=
```
Comment (monit-reviewer) [inline tools/jtk/go.sum:38]: 🔵 **Low** (harness-engineering:harness-knowledge-reviewer): github.com/clipperhouse/stringish v0.1.1 and github.com/clipperhouse/uax29/v2 v2.3.0 are present in go.sum with only module-zip h1: hashes; the corresponding /go.mod h1: hash lines are absent. Standard go mod tidy produces both. Missing go.mod hashes cause go mod verify to fail and indicate the lockfile was not generated by a clean tidy run.

*Reply to this thread when addressed.*

**Review by monit-reviewer — DISMISSED** [review]
## Automated PR Review



**Reviewed commit:** `6085d65` | **Previous:** `7d88fcc` (incremental)

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-architecture-reviewer | 1 |
| harness-engineering:harness-enforcement-reviewer | 2 |
| harness-engineering:harness-knowledge-reviewer | 1 |

<details>
<summary><strong>harness-engineering:harness-architecture-reviewer (1 findings)</strong></summary>

**⚠️ Should Fix** - `tools/cfl/go.mod:1`
> tools/cfl/go.mod was not updated to require github.com/open-cli-collective/atlassian-go (the shared atime package), even though tools/cfl/api/types.go now imports it. The module builds correctly inside the Go workspace via go.work, but the go.mod is incomplete and will fail standalone non-workspace builds or a clean go mod tidy run.

</details>

<details>
<summary><strong>harness-engineering:harness-enforcement-reviewer (2 findings)</strong></summary>

**💡 Suggestion** - `shared/atime/atime.go:63`
> The inline comment states 'epoch millis are 13+ digits for dates after 2001', but the code routes any value with len > 10 digits to UnixMilli, meaning 11- and 12-digit values are also treated as millis. The comment is misleading: the actual threshold is 11+ digits, not 13+. A future maintainer reading only the comment may misunderstand the heuristic.

**💡 Suggestion** - `shared/atime/atime_test.go:143`
> The contains() and searchString() helpers manually reimplement strings.Contains. Use strings.Contains from the standard library directly.

</details>

<details>
<summary><strong>harness-engineering:harness-knowledge-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/go.sum:38`
> github.com/clipperhouse/stringish v0.1.1 and github.com/clipperhouse/uax29/v2 v2.3.0 are present in go.sum with only module-zip h1: hashes; the corresponding /go.mod h1: hash lines are absent. Standard go mod tidy produces both. Missing go.mod hashes cause go mod verify to fail and indicate the lockfile was not generated by a clean tidy run.

</details>

*1 info-level observations excluded. Run with `--verbose` to include.*

*9 PR discussion threads considered.*

---
<details>
<summary>Completed in 5m 04s | $1.33</summary>

| Field | Value |
|---|---|
| Mode | Re-review · Cycle 2 · Session resumed |
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 5m 04s wall · 12m 27s compute (Reviewers: 3m 54s · Synthesis: 45s) |
| Cost | $1.33 |
| Tokens | 425.8k in / 41.2k out |
| Turns | 18 |

</details>

*Note: Posted 3 inline comments. 1 finding remained in the summary because GitHub could not attach them to the current diff.*

File: shared/atime/atime.go (hunk 7 above)
Comment (rianjs) [reply shared/atime/atime.go:63]: Fixed — improved comment to clarify 11-12 digit values are ambiguous but don't occur in Atlassian API responses.

File: shared/atime/atime_test.go (hunk 8 above)
Comment (rianjs) [reply shared/atime/atime_test.go:144]: Fixed — replaced custom contains/searchString with strings.Contains.

File: tools/jtk/go.sum (hunk 9 above)
Comment (rianjs) [reply tools/jtk/go.sum:38]: Not actionable — go.sum hash format is controlled by the Go toolchain, not the project.

File: shared/atime/atime.go (hunk 7 above)
Comment (monit-reviewer) [reply shared/atime/atime.go:63]: **Settled**: monit-reviewer flagged a misleading inline comment claiming epoch millis are 13+ digits when the actual threshold is 11+. rianjs fixed the comment to clarify that 11-12 digit values are ambiguous but don't appear in Atlassian API responses.

File: shared/atime/atime_test.go (hunk 8 above)
Comment (monit-reviewer) [reply shared/atime/atime_test.go:144]: **Settled**: monit-reviewer flagged custom contains() and searchString() helpers as redundant reimplementations of strings.Contains. rianjs replaced them with strings.Contains from the standard library.

File: tools/jtk/go.sum (hunk 9 above)
Comment (monit-reviewer) [reply tools/jtk/go.sum:38]: **Settled**: monit-reviewer flagged missing /go.mod h1: hash lines in go.sum for two dependencies. rianjs explained this is not actionable as go.sum hash format is controlled by the Go toolchain, not the project.

File: tools/jtk/internal/present/automation.go [hunk 10]
```diff
@@ -385,3 +386,10 @@ func SummarizeComponents(components []api.RuleComponent) string {
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/automation.go:386]: 🟡 **Medium** (harness-engineering:harness-self-documenting-code-reviewer): formatAtlassianTimeOrDash calls FormatDate(&t.Time), but the old code called FormatTime(string(rule.Created)). If FormatDate produces only a YYYY-MM-DD string while FormatTime produced a full datetime, the 'Created / Updated' line in the extended get view silently lost time-of-day information. The test in get_test.go only asserts a date prefix ("Created: 2023-12-02") so this regression would not be caught. Confirm that date-only display is intentional, or switch to FormatTime with the new type.

*Reply to this thread when addressed.*

File: shared/atime/atime.go [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+func (t *AtlassianTime) UnmarshalJSON(data []byte) error {
+	if len(data) == 0 || string(data) == "null" {
+		return nil
+	}
+
+	if data[0] == '"' {
+		s := strings.Trim(string(data), `"`)
+		if s == "" {
```
Comment (monit-reviewer) [inline shared/atime/atime.go:36]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): strings.Trim(string(data), `"`) strips ALL leading and trailing quote characters rather than just the outer pair. Malformed JSON like `""foo""` would silently produce `foo` instead of an error. The idiomatic approach is to confirm data[0] == '"' && data[len(data)-1] == '"' then slice, or unmarshal into a string via json.Unmarshal.

*Reply to this thread when addressed.*

File: shared/atime/atime_test.go [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+			name:    "unparseable ISO string",
+			input:   `"not-a-date"`,
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			var at AtlassianTime
```
Comment (monit-reviewer) [inline shared/atime/atime_test.go:83]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): No test covers the 11- or 12-digit integer boundary that the implementation comment explicitly calls out as ambiguous. A test asserting the current (millis) interpretation for an 11-digit input would document intended behavior and prevent silent regression if the heuristic is ever revised.

*Reply to this thread when addressed.*

**Review by monit-reviewer — DISMISSED** [review]
## Automated PR Review



**Reviewed commit:** `ed5db94` | **Previous:** `6085d65` (incremental)

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-self-documenting-code-reviewer | 3 |

<details>
<summary><strong>harness-engineering:harness-self-documenting-code-reviewer (3 findings)</strong></summary>

**⚠️ Should Fix** - `tools/jtk/internal/present/automation.go:386`
> formatAtlassianTimeOrDash calls FormatDate(&t.Time), but the old code called FormatTime(string(rule.Created)). If FormatDate produces only a YYYY-MM-DD string while FormatTime produced a full datetime, the 'Created / Updated' line in the extended get view silently lost time-of-day information. The test in get_test.go only asserts a date prefix ("Created: 2023-12-02") so this regression would not be caught. Confirm that date-only display is intentional, or switch to FormatTime with the new type.

**💡 Suggestion** - `shared/atime/atime.go:36`
> strings.Trim(string(data), `"`) strips ALL leading and trailing quote characters rather than just the outer pair. Malformed JSON like `""foo""` would silently produce `foo` instead of an error. The idiomatic approach is to confirm data[0] == '"' && data[len(data)-1] == '"' then slice, or unmarshal into a string via json.Unmarshal.

**💡 Suggestion** - `shared/atime/atime_test.go:83`
> No test covers the 11- or 12-digit integer boundary that the implementation comment explicitly calls out as ambiguous. A test asserting the current (millis) interpretation for an 11-digit input would document intended behavior and prevent silent regression if the heuristic is ever revised.

</details>

*5 info-level observations excluded. Run with `--verbose` to include.*

*12 PR discussion threads considered.*

---
<details>
<summary>Completed in 4m 05s | $1.45</summary>

| Field | Value |
|---|---|
| Mode | Re-review · Cycle 3 · Session resumed |
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 4m 05s wall · 11m 14s compute (Reviewers: 2m 32s · Synthesis: 1m 18s) |
| Cost | $1.45 |
| Tokens | 429.8k in / 39.1k out |
| Turns | 17 |

</details>

File: tools/jtk/internal/present/automation.go (hunk 10 above)
Comment (rianjs) [reply tools/jtk/internal/present/automation.go:386]: No behavior change — FormatTime(string) also returns only the date portion (it truncates to first 10 chars, i.e. YYYY-MM-DD). FormatDate produces the same YYYY-MM-DD output. The old and new paths are functionally identical.

File: shared/atime/atime.go (hunk 11 above)
Comment (rianjs) [reply shared/atime/atime.go:36]: Accepted risk — JSON strings always have exactly one leading and one trailing quote. Malformed JSON like '""foo""' would never reach UnmarshalJSON since json.Decoder rejects it first. The Trim approach is idiomatic for this pattern in Go JSON unmarshalers.

File: shared/atime/atime_test.go (hunk 12 above)
Comment (rianjs) [reply shared/atime/atime_test.go:83]: Intentional omission — 11-12 digit values are explicitly called out as ambiguous in the comment and don't occur in Atlassian API responses. Testing them would codify behavior for an input that shouldn't exist, creating a false sense of coverage.

**Comment by rianjs** [comment]
These are low-value, please approve the PR

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `d83e451`

### Human Override

Approved at the request of @rianjs ([comment](https://github.com/open-cli-collective/atlassian-cli/pull/310#issuecomment-4333947917)).

> These are low-value, please approve the PR

No automated reviewers were run.
